src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -52,90 +55,259 @@ public Class<CreateCollectionCommand> getCommandClass() {
+   * @param hasVectorSearch indicating if vector search options are enabled.
+   * @param commandName command name
+   * @param indexing the indexing option config
+   * @param vector the vector option config
+   * @return the comment string
+   */
+  private String generateComment(
+      boolean hasIndexing,
+      boolean hasVectorSearch,
+      String commandName,
+      CreateCollectionCommand.Options.IndexingConfig indexing,
+      CreateCollectionCommand.Options.VectorSearchConfig vector) {
+    if (!hasIndexing && !hasVectorSearch) {
+      return null;
+    }
+    final ObjectNode collectionNode = objectMapper.createObjectNode();
+    ObjectNode optionsNode = objectMapper.createObjectNode(); // For storing collection options.
+
+    if (hasIndexing) {
+      optionsNode.putPOJO(TableCommentConstants.COLLECTION_INDEXING_KEY, indexing);
+    }
+    if (hasVectorSearch) {
+      optionsNode.putPOJO(TableCommentConstants.COLLECTION_VECTOR_KEY, vector);
+    }
+
+    collectionNode.put(TableCommentConstants.COLLECTION_NAME_KEY, commandName);
+    collectionNode.put(
+        TableCommentConstants.SCHEMA_VERSION_KEY, TableCommentConstants.SCHEMA_VERSION_VALUE);
+    collectionNode.putPOJO(TableCommentConstants.OPTIONS_KEY, optionsNode);
+    final ObjectNode tableCommentNode = objectMapper.createObjectNode();
+    tableCommentNode.putPOJO(TableCommentConstants.TOP_LEVEL_KEY, collectionNode);
+
+    return tableCommentNode.toString();
+  }
+
+  private CreateCollectionCommand.Options.VectorSearchConfig validateVectorOptions(
+      CreateCollectionCommand.Options.VectorSearchConfig vector) {
+    String function = vector.metric();
+    Integer vectorDimension = vector.dimension();
+    CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig service =
+        vector.vectorizeConfig();
+
+    if (service != null) {
+      // Validate service configuration and auto populate vector dimension.
+      vectorDimension = validateService(service, vectorDimension);
+      vector =
+          new CreateCollectionCommand.Options.VectorSearchConfig(
+              vectorDimension, vector.metric(), vector.vectorizeConfig());
+    } else {
+      // Ensure vector dimension is provided when service configuration is absent.
+      if (vectorDimension == null) {
+        throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+            "The dimensions can not be null if service is not provided");
+      }
+    }
+    // TODO: what if the config model vector dimension is larger than 4096?
+    if (vectorDimension > documentLimitsConfig.maxVectorEmbeddingLength()) {
+      throw new JsonApiException(
+          ErrorCode.VECTOR_SEARCH_TOO_BIG_VALUE,
+          String.format(
+              "%s: %d (max %d)",
+              ErrorCode.VECTOR_SEARCH_TOO_BIG_VALUE.getMessage(),
+              vectorDimension,
+              documentLimitsConfig.maxVectorEmbeddingLength()));
+    }
```

Comment:
Will we support the model that generates a vector dimension larger than 4096 (`maxVectorEmbeddingLength`)?

Reply:
DB currently can't support vectors > 4096